dnf/cli/commands/repoquery.py

```diff
@@ -78,6 +82,46 @@ def build_format_fn(opts):
         return rpm2py_format(opts.queryformat).format
 
 
+def _enable_sub_repos(repos, sub_name_fn):
+    for repo in repos.iter_enabled():
+        for found in repos.get_matching(sub_name_fn(repo.id)):
+            if not found.enabled:
+                logger.info(_('enabling %s repository'), found.id)
+                found.enable()
+
+
+def enable_source_repos(repos):
+    """
+    enable source repos corresponding to already enabled binary repos
+    """
+    def source_name(name):
+        return ("{}-source-rpms".format(name[:-5]) if name.endswith("-rpms")
+                else "{}-source".format(name))
+    _enable_sub_repos(repos, source_name)
+
+
+def package_source_name(package):
+    """"
+    returns name of source package for given pkgname
+    e.g. krb5-libs -> krb5
+    """
+    if package.sourcerpm is not None:
+        # trim suffix first
+        srcname = rtrim(package.sourcerpm, ".src.rpm")
+        # source package filenames may not contain epoch, handle both cases
+        srcname = rtrim(srcname, "-{}".format(package.evr))
+        srcname = rtrim(srcname, "-{0.version}-{0.release}".format(package))
+    else:
+        srcname = None
+    return srcname
+
+
+def rtrim(s, r):
+    while s.endswith(r):
+        s = s[:-len(r)]
+    return s
```

Comment:
I'd prefer to have these function in e.g dnf/utils.py so dnf-plugins-core can re-use them rather tha duplicate.